client/signup/steps/jpo-site-type/select-business-address.jsx

```diff
@@ -1,45 +1,82 @@
+/**
+ * External dependencies
+ */
 import React from 'react';
-
+import PropTypes from 'prop-types';
 import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
 import Card from 'components/card';
 import Button from 'components/button';
 import FormLabel from 'components/forms/form-label';
 import FormTextInput from 'components/forms/form-text-input';
-import FormTextarea from 'components/forms/form-textarea';
-
-import PersonalGraphic from './personal-graphic';
-import BusinessGraphic from './business-graphic';
 
-module.exports = React.createClass( {
+class SelectBusinessAddress extends React.Component {
 
-	displayName: 'SelectBusinessAddress',
+	static propTypes = {
+		signupDependencies: PropTypes.object,
+		handleBusinessInfo: PropTypes.func,
+		businessInfo: PropTypes.object,
+		required: PropTypes.bool,
+	};
 
-	propTypes: {
-		required: React.PropTypes.bool,
-	},
-
-	render: function() {
+	render() {
 		if ( ! this.props.current ) {
 			return ( <div /> );
 		}
 
+		const {
+			handleBusinessInfo,
+			businessInfo: {
+				businessName,
+				streetAddress,
+				city,
+				state,
+				zipCode
+			}
+		} = this.props;
+
 		return ( 
 			<div className="jpo__site-type-wrapper business-address">
 				<Card>
 					<FormLabel>{ translate( 'Business Name' ) }</FormLabel>
-					<FormTextInput onChange={ this.props.onInputBusinessName } />
+					<FormTextInput
+						name="businessName"
+						value={ businessName }
+						onChange={ handleBusinessInfo }
+						/>
 					<FormLabel>{ translate( 'Street Address' ) }</FormLabel>
-					<FormTextInput onChange={ this.props.onInputStreetAddress } />
+					<FormTextInput
+						name="streetAddress"
+						value={ streetAddress }
+						onChange={ handleBusinessInfo }
+						/>
 					<FormLabel>{ translate( 'City' ) }</FormLabel>
-					<FormTextInput onChange={ this.props.onInputCity } />
+					<FormTextInput
+						name="city"
+						value={ city }
+						onChange={ handleBusinessInfo }
+						/>
 					<FormLabel>{ translate( 'State' ) }</FormLabel>
-					<FormTextInput onChange={ this.props.onInputState } />
+					<FormTextInput
+						name="state"
+						value={ state }
+						onChange={ handleBusinessInfo }
+						/>
 					<FormLabel>{ translate( 'ZIP Code' ) }</FormLabel>
```

Comment:
Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Zip Code' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/jpo-selectable-blocks)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).